src/main/cpp/threadutility.cpp

```diff
@@ -44,19 +50,23 @@ void ThreadUtility::configureFuncs( ThreadStartPre pre_start,
 
 void ThreadUtility::preThreadBlockSignals(){
 #if LOG4CXX_HAS_PTHREAD_SIGMASK
+	m_priv->creation_mutex.lock();
```

Comment:
> Thorsten's compiler doesn't support it,
(Thread local is required to be supported by compiler claiming C++11 compliance; and https://logging.apache.org/log4cxx/latest_stable/dependencies.html says log4cxx requires C++11 or later....)

I agree, as Signals is a Unix thing, ignoring it might be it as well.

Another (Unix) approach could be Thread-Specific-Data (pthread_key_create(), pthread_setspecific() and friends)
Disclaimer: Never used those functions myself until now (as thread local storage is far easier to use and has less overhead).

Thinking... Another idea would be around avoiding that singleton, or at least using the singleton in a different way...
mmm.... 

Maybe a factory instead of the singleton? It could returns individual IThreadFunctions-derived objects and thus having its own members variables

The factory would be configured by the Configure() function; 
Rather than having individual functions to be set with configureFuncs() I'd encapsulate the 3 functions in in the class/struct IThreadFunctions. IThreadFunctions could have a (maybe static) clone() function to help the factory.

(OTOH .. .I'm still not sure if this configurabiity/complixty is really needed; Possibly it would be enough to block the signals unconditionally when log4cxx starts a thread?  I currently cant think of an usecase where a user would not want that log4cxx just blocks the signals. (Please explain; I'm obviously missing why you think it necessary)..)

Reply:
> (OTOH .. .I'm still not sure if this configurabiity/complixty is really needed; Possibly it would be enough to block the signals unconditionally when log4cxx starts a thread? I currently cant think of an usecase where a user would not want that log4cxx just blocks the signals. (Please explain; I'm obviously missing why you think it necessary)..)

My general philosophy is that sane defaults should be acceptable >90% of the time, but there are some instances where you might want to customize this for whatever reason.  Hence the single-line call to ThreadUtility::configure to set various default configurations, while still allowing the end-user to customize.  It's unlikely, but users could run on an OS that uses some sort of threading implementation that doesn't use pthreads, but still needs to do something before the thread starts.

Part of the 'naming threads' as well is that there are some systems where you can name threads and some systems where you can't - Linux is [pthread_setname_np](https://man7.org/linux/man-pages/man3/pthread_setname_np.3.html), BSD may be [pthread_set_name_np](https://www.freebsd.org/cgi/man.cgi?query=pthread_set_name_np), and OSX uses pthread_setname_np but only [in the current thread](https://opensource.apple.com/source/libpthread/libpthread-301.30.1/pthread/pthread.h.auto.html).

I don't particularly care if the blocking of signals is off or on by default, as long as there is a way to change it.

Reply:
Well, pthreads is POSIX, so it should be quite portable.
(Obviously - the name, but that seldom see anyone set the name of a thread)

Yes, I suggest that blocking the signals should be the default

Reply:
I'm going to merge this for now, and do a separate PR in a little bit that configures the blocking of signals to be set by default.  There are a few things that I want to take a look at in terms of how/when that gets configured.

Reply:
:+1: 

> There are a few things that I want to take a look at in terms of how/when that gets configured.

Yes, that would be interesting informatin. One thing that would explcitly interest me is "when are the threads started". (To assess when liblog4cxx spawns it threads and therefore fork() becoming unsafe without exec())